error in new build_ext

@rainwoodman I'm seeing errors in the new build_ext from PR #64 

It looks like the .so extension is not being built.  You might not see it if
you haven't removed the /build directory

```
building 'fitsio._fitsio_wrap' extension
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/fitsio
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/esheldon/anaconda/envs/local/lib/python2.7/site-packages/numpy/core/include -I/home/esheldon/anaconda/envs/local/include/python2.7 -Icfitsio3370patch -c fitsio/fitsio_pywrap.c -o build/temp.linux-x86_64-2.7/fitsio/fitsio_pywrap.o
In file included from /home/esheldon/anaconda/envs/local/include/python2.7/Python.h:8:0,
                 from fitsio/fitsio_pywrap.c:24:
/home/esheldon/anaconda/envs/local/include/python2.7/pyconfig.h:1188:0: warning: "_POSIX_C_SOURCE" redefined [enabled by default]
 #define _POSIX_C_SOURCE 200112L
 ^
In file included from /usr/include/string.h:25:0,
                 from fitsio/fitsio_pywrap.c:23:
/usr/include/features.h:230:0: note: this is the location of the previous definition
 # define _POSIX_C_SOURCE 200809L
 ^
In file included from /home/esheldon/anaconda/envs/local/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1804:0,
                 from /home/esheldon/anaconda/envs/local/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:17,
                 from /home/esheldon/anaconda/envs/local/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from fitsio/fitsio_pywrap.c:28:
/home/esheldon/anaconda/envs/local/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^
gcc -pthread -shared -L/home/esheldon/anaconda/envs/local/lib -Wl,-rpath=/home/esheldon/anaconda/envs/local/lib,--no-as-needed build/temp.linux-x86_64-2.7/fitsio/fitsio_pywrap.o build/cfitsio3370patch/libcfitsio.a -L/home/esheldon/anaconda/envs/local/lib -lpython2.7 -o build/lib.linux-x86_64-2.7/fitsio/_fitsio_wrap.so
/usr/bin/ld: cannot open output file build/lib.linux-x86_64-2.7/fitsio/_fitsio_wrap.so: No such file or directory
collect2: error: ld returned 1 exit status
error: command 'gcc' failed with exit status 1
```

--- a/setup.py
+++ b/setup.py
@@ -64,9 +64,8 @@ class build_ext_subclass(build_ext):
             # when using "extra_objects" in Extension, changes in the objects do *not*
             # cause a re-link!  The only way I know is to force a recompile by removing the
             # directory
-            build_libdir=glob.glob(os.path.join('build','lib*'))
-            if len(build_libdir) > 0:
-                shutil.rmtree(build_libdir[0])
+            for sofile in glob.glob(os.path.join('build','lib*/fitsio/*_fitsio_wrap*.so*')):
+                os.unlink(sofile)
 
             # link against the .a library in cfitsio; 
             # It should have been a 'static' library of relocatable objects (-fPIC), 

